src/services/reviews.js

```diff
@@ -0,0 +1,447 @@
+					//approve the resource
+					isPublishResource = await this.handleApproval(
+						review.id,
+						resourceId,
+						loggedInUserId,
+						bodyData.comment,
+						minApproval,
+						resource.organization_id,
+						review.organization_id
+					)
+				}
+			}
+
+			// Publish resource if applicable
+			if (isPublishResource) {
+				const publishResource = await resourceService.publishResource(resourceId, resourceDetails.user_id)
+				return publishResource
+			}
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'REVIEW_UPDATED',
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	/**
+	 * Reject/Report the resource
+	 * @method
+	 * @name handleRejectionOrReport
+	 * @param {Integer} reviewId - review id.
+	 * @param {Object} bodyData - review body data.
+	 * @param {Integer} resourceId - resource id.
+	 * @param {String} loggedInUserId - logged in user id.
+	 * @param {Array} comments - comments.
+	 * @param {String} resourceType - resource type.
+	 * @param {String} resourceOrgId - resource org id.
+	 * @param {String} reviewOrgId - review org id.
+	 * @returns {JSON} - review updated response.
+	 */
+	static async handleRejectionOrReport(
+		reviewId,
+		bodyData,
+		resourceId,
+		loggedInUserId,
+		comments = [],
+		resourceType,
+		resourceOrgId,
+		reviewOrgId
+	) {
+		try {
+			//program cannot reject or report
+			if (resourceType === common.RESOURCE_TYPE_PROGRAM) {
+				return responses.failureResponse({
+					message: 'PROGRAM_REJECTION_NOT_ALLOWED',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			// handle rejection or report of a review
+			let updateObj = { status: bodyData.status }
+			if (bodyData.notes) {
+				updateObj.notes = bodyData.notes
+			}
+
+			await resourceQueries.updateOne(
+				{ id: resourceId, organization_id: resourceOrgId },
+				_.omit(updateObj, ['notes'])
+			)
+
+			await reviewsQueries.update({ id: reviewId, organization_id: reviewOrgId }, updateObj)
+
+			// Add or update comments
+			if (comments) {
+				await handleComments(comments, resourceId, loggedInUserId)
+			}
+
+			const message =
+				bodyData.status === common.REVIEW_STATUS_REJECTED ? 'REVIEW_REJECTED' : 'REVIEW_REJECTED_AND_REPORTED'
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message,
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	/**
+	 * Update the review
+	 * @method
+	 * @name handleUpdateReview
+	 * @param {Integer} reviewId - review id.
+	 * @param {Object} bodyData - review body data.
+	 * @param {Integer} resourceId - resource id.
+	 * @param {String} loggedInUserId - logged in user id.
+	 * @param {Array} comments - comments.
+	 * @param {String} organization_id - org id.
+	 * @returns {JSON} - review updated response.
+	 */
+	static async handleUpdateReview(reviewId, bodyData, resourceId, loggedInUserId, comments = [], organization_id) {
```

Comment:
@priyanka-TL As discussed we will go with 3 different APIs - approve/update/reject&report